### PR TITLE
[DOCS] Rename Flink Table Store to Apache Paimon (Incubating) in docs `Connectors for Flink SQL Query Engine`

### PR DESCRIPTION
**Pull request by huage1994**
### _Why are the changes needed?_

To update docs.
Rename Flink Table Store to Apache Paimon (Incubating) in docs `Connectors for Flink SQL Query Engine`

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by SteNicholas — APPROVED** [review]
LGTM. cc @pan3793

File: docs/connector/flink/paimon.rst [hunk 1]
```diff
@@ -13,57 +13,57 @@
    See the License for the specific language governing permissions and
    limitations under the License.
 
-`Flink Table Store`_
+`Apache Paimon (Incubating)`_
 ==========
```
Comment (yaooqinn) [inline docs/connector/flink/paimon.rst:17]: padding with enough '='

File: docs/connector/flink/paimon.rst [hunk 2]
```diff
@@ -13,57 +13,57 @@
...
    you can obtain it from its `Official Documentation`_.
 
-By using kyuubi, we can run SQL queries towards Flink Table Store which is more
+By using kyuubi, we can run SQL queries towards Apache Paimon (Incubating) which is more
 convenient, easy to understand, and easy to expand than directly using
-flink to manipulate Flink Table Store.
+flink to manipulate Apache Paimon (Incubating).
 
-Flink Table Store Integration
+Apache Paimon (Incubating) Integration
```
Comment (yaooqinn) [inline docs/connector/flink/paimon.rst:30]: `Apache Paimon (Incubating)` might be redundant

File: docs/connector/flink/paimon.rst [hunk 3]
```diff
@@ -13,57 +13,57 @@
...
-1. Put the Flink Table Store packages into ``$FLINK_HOME/lib`` directly
+1. Put the Apache Paimon (Incubating) packages into ``$FLINK_HOME/lib`` directly
 2. Setting the HADOOP_CLASSPATH environment variable or copy the `Pre-bundled Hadoop Jar`_ to flink/lib.
 
 .. warning::
-   Please mind the compatibility of different Flink Table Store and Flink versions, which can be confirmed on the page of `Flink Table Store multi engine support`_.
+   Please mind the compatibility of different Apache Paimon (Incubating) and Flink versions, which can be confirmed on the page of `Apache Paimon (Incubating) multi engine support`_.
 
-Flink Table Store Operations
+Apache Paimon (Incubating) Operations
```
Comment (yaooqinn) [inline docs/connector/flink/paimon.rst:57]: ditto

File: docs/connector/flink/paimon.rst [hunk 4]
```diff
@@ -13,57 +13,57 @@
    See the License for the specific language governing permissions and
    limitations under the License.
 
-`Flink Table Store`_
+`Apache Paimon (Incubating)`_
 ==========
 
-Flink Table Store is a unified storage to build dynamic tables for both streaming and batch processing in Flink,
-supporting high-speed data ingestion and timely data query.
+Apache Paimon(incubating) is a streaming data lake platform that supports high-speed data ingestion, change data tracking and efficient real-time analytics.
```
Comment (yaooqinn) [inline docs/connector/flink/paimon.rst:19]: use `Apache Paimon (Incubating)` like other places

**Comment by huage1994** [comment]
Thanks @yaooqinn for the review. I have modified the docs. It looks like there are no errors or warning in the compiled output

**Review by yaooqinn — APPROVED** [review]

File: docs/connector/flink/paimon.rst [hunk 5]
```diff
@@ -13,57 +13,56 @@
...
    limitations under the License.
 
-`Flink Table Store`_
-==========
+`Apache Paimon (Incubating)`_
+=============================
 
-Flink Table Store is a unified storage to build dynamic tables for both streaming and batch processing in Flink,
-supporting high-speed data ingestion and timely data query.
+Apache Paimon (incubating) is a streaming data lake platform that supports high-speed data ingestion, change data tracking and efficient real-time analytics.
```
Comment (yaooqinn) [inline docs/connector/flink/paimon.rst:19]:
```suggestion
Apache Paimon (Incubating) is a streaming data lake platform that supports high-speed data ingestion, change data tracking, and efficient real-time analytics.
```

**Comment by pan3793** [comment]
Thanks, merged to master/1.7

**Pull request closed without merging**